### ISSUE TITLE
Writing_Linter_Bears docs for global-linter-bears produce TypeError: `self` is missing in `create_arguments` signature

### ISSUE THREAD

**Issue by sangamcse** [coala/coala#5354]
In [Writing_Linter_Bears.html#global-linter-bears](http://api.coala.io/en/latest/Developers/Writing_Linter_Bears.html#global-linter-bears), it is written `create_arguments(config_file)`, but it gives TypeError written below.
```
TypeError: create_arguments() takes 1 positional argument but 2 were given
```

status/needs discussion

**Comment by sachinmukherjee** [comment]
@jayvdb can I fix this issue?

**Comment by sangamcse** [comment]
@sachinmukherjee, this issue is yet to be discussed. Can you find another issue?

**Comment by sachinmukherjee** [comment]
@sangamcse ok sir

**Comment by Makman2** [comment]
`self` is missing in the signature:
```python
def create_arguments(self, config_file)
```

**Comment by z1ggy-o** [comment]
Can I work on this issue?

**Comment by jayvdb** [comment]
Welcome aboard @z1ggy-o

**Comment by jayvdb** [comment]
@sangamcse , can one of you give the required arguments for this here, and on the PR.

**Comment by sangamcse** [comment]
@jayvdb, it looks like I did a mistake while checking the docs. The listed issue https://github.com/coala/coala/issues/5354#issue-312658505 occurred when I am not using `@staticmethod`(which previously I did not cross check while going through the docs). When I use `@staticmethod`, it gives different error --> 
```
[ERROR][01:58:09] The given arguments None are not iterable.
```
which IMO is not an issue of the docs.
Thanks to @palash25.

**Comment by sangamcse** [comment]
In case this issue is closed, I have a newcomer task for @z1ggy-o in the same docs. Feel free to ping.

**Comment by z1ggy-o** [comment]
@sangamcse appreciate it! 
But I should wait for the result of this issue before starting another newcomer task, right?

**Comment by jayvdb** [comment]
I think @palash25 is onto something.  `CPPCheckBear.create_arguments` uses `self.file_dict` to get the list of all files.  It is not a `@staticmethod`.

So our docs here should at least also say that `CPPCheckBear.create_arguments` can be a normal method, with `self` as the first parameter, and indicate that is a way to access the complete list of files using `self.file_dict`.

This should come with a warning that putting the complete list of files on the command line will cause breakages when the length of the command line exceeds the OS permitted length, or the complete list of files is greater than the OS permitted number of args.  See https://www.in-ulm.de/~mascheck/various/argmax/ for info about that.

@sangamcse , if the example in the docs gives an error, it isnt a very good example.  We should make this work correctly.  Perhaps instead of `pass` it should be `return []`.

ping @Makman2 again.

**Comment by Makman2** [comment]
> self is missing in the signature:
>
> def create_arguments(self, config_file)

My bad I've missed the `@staticmethod` there. However this errors usually appears if you forget to annotate your `create_argument` function correctly to match with your params ;)

**Comment by Makman2** [comment]
And yeah, we should `return []` there :+1:

**Comment by manankalra** [comment]
>So our docs here should at least also say that CPPCheckBear.create_arguments can be a normal method, with self as the first parameter, and indicate that is a way to access the complete list of files using self.file_dict.

If ^this is regarding `Writing_Linter_Bears.rst`, then it already contains a note stating:
>create_arguments doesn’t have to be a static method. In this case you also need to prepend self to the parameters in the signature ... ...

Maybe this note can be changed to include a relevant example that demonstrates the use of a `file_dict`.
